trace-dispatcher/doc/trace-dispatcher.md

```diff
@@ -108,689 +89,654 @@ data TraceAddBlockEvent blk =
+docuResultsToText :: DocTracer -> TraceConfig -> IO Text
 
-```yaml
-  Node.BlockFetchClient:
-    maxFrequency: 1.0
+-- For example
+  b1 <- documentTracer traceForgeEventDocu [t1, t2]
+  b2 <- documentTracer .. ..
+  ..
+  bn <- documentTracer .. ..
+  writeFile "Docu.md" (docuResultsToText (b1 ++ b2 ++ ... ++ bn))
 ```
 
-In Cardano a default configuration is given in the module [Cardano.Node.Tracing.DefaultTraceConfig](https://github.com/intersectmbo/cardano-node/blob/master/cardano-node/src/Cardano/Node/Tracing/DefaultTraceConfig.hs). In the config file all entries of the default configuration can be overridden. To remove a frequency limiter, define a limiter with maxFrequency 0.0.
-
-## Documentation
+A generated documentation snippet for a simple message may appear as follows:
 
-The self-documentation features of `trace-dispatcher` are provided by a combination of:
+__Forge.Loop.StartLeadershipCheck__
 
-* __documentation annotations__, expressed by the `Documented a` type, that carry a list of per- __message__ description (of `DocMsg` type), and
-* a special __dispatcher__ execution mode that emits documentation for all annotated traces, using the __tracer namespace__ to guide the document structure.
+> Start of the leadership check.
 
-The per- __message__ `DocMsg` objects combine:
+Severity:  `Info`
+Privacy:   `Public`
+Details:   `DNormal`
 
-* `namespace` of the message
-* Message documentation text, in Markdown format
-* A list of tuples with metrics names and metrics documentation
+From the current configuration:
 
-If a documentation entry is missing, the configuration system will not work as expected.
-Because it is not enforced by the type system, it is very important that each trace documentation provides a complete list of `DocMsg` entries for all message contructors, as these entries are also used for configuration. If a message has no DocMsg, it can't be configured individually.
+Backends:
+      `EKGBackend`,
+      `Stdout MachineFormat`,
+      `Forwarder`
+Filtered `Visible` by config value: `Info`
 
-TODO: Add consistency check between configuration and documentation.
+## Consistency Checking
 
+As namespaces are essentially strings, the type system doesn't inherently ensure the consistency of namespaces. To address this concern, we have incorporated consistency check functionality into `trace-dispatcher`. Within the node, you can invoke the following procedure from the `Cardano.Node.Tracing.Consistency` module. It returns an array of `Text`, an empty list indicating that everything is in order.
 
 ```haskell
-newtype Documented a = Documented {undoc :: [DocMsg a]}
-
-data DocMsg a = DocMsg {
-    dmNamespace :: NamespaceOuter
-  , dmMetricsMD :: [(Text, Text)]
-  , dmMarkdown  :: Text
-}
+-- | Check the configuration in the given file.
+-- Check the general structure of namespaces.
+-- An empty return list means everything is well.
+checkNodeTraceConfiguration ::
+     FilePath
+  -> IO [Text]
```

Comment:
Is the argument supposed to be a node config file? What would be an example of a non-empty result (i.e. a failed consistency check)?

Reply:
Again yes, a node config file. I added: An example text is "Config namespace error: i.am.an.invalid.namepace".